quality_control_oca/models/qc_inspection.py

```diff
@@ -119,6 +123,14 @@ def create(self, val_list):
                 vals["name"] = self.env["ir.sequence"].next_by_code("qc.inspection")
         return super().create(vals)
 
+    def write(self, vals):
+        if "state" in vals:
+            if vals["state"] in ["success", "failed"]:
+                vals["date_done"] = datetime.now()
+            elif vals["state"] == "draft":
+                vals["date_done"] = False
+        return super().write(vals)
```

Comment:
Can you please update tests to check on `date_done` field after relevant actions?